src/mod_mam_sql.erl

```diff
@@ -117,42 +118,90 @@ store(Pkt, LServer, {LUser, LHost}, Type, Peer, Nick, _Dir, TS) ->
 		  fxml:element_to_binary(Pkt)
 	  end,
 	case SqlType of
-	  mssql -> case ejabberd_sql:sql_query(
-	           LServer,
-	           ?SQL_INSERT(
-	              "archive",
-	              ["username=%(SUser)s",
-	               "server_host=%(LServer)s",
-	               "timestamp=%(TS)d",
-	               "peer=%(LPeer)s",
-	               "bare_peer=%(BarePeer)s",
-	               "xml=N%(XML)s",
-	               "txt=N%(Body)s",
-	               "kind=%(SType)s",
-	               "nick=%(Nick)s"])) of
-		{updated, _} ->
-		    ok;
-		Err ->
-		    Err
-	    end;
-	    _ -> case ejabberd_sql:sql_query(
-	           LServer,
-	           ?SQL_INSERT(
-	              "archive",
-	              ["username=%(SUser)s",
-	               "server_host=%(LServer)s",
-	               "timestamp=%(TS)d",
-	               "peer=%(LPeer)s",
-	               "bare_peer=%(BarePeer)s",
-	               "xml=%(XML)s",
-	               "txt=%(Body)s",
-	               "kind=%(SType)s",
-	               "nick=%(Nick)s"])) of
-		{updated, _} ->
-		    ok;
-		Err ->
-		    Err
-	    end
+	  mssql ->
```

Comment:
This case is irrelevant in our case, but I included it for the sake of completeness.